### PR TITLE
Stats: Adds a banner when the post has no views yet

### LINKED ISSUE #10381: Stats: Post Summary for posts without stats is Sad
When viewing a post summary stats page for a post that has no visitor stats, things look a bit sad:

wordpress_com

Instead of showing empty tables dating back to 1970, we should display a more friendly message stating the post has yet to receive views.  Perhaps we could encourage publicize on this screen too if no connections are setup on the site yet, and/or link to documentation on how to increase traffic.

### PR DESCRIPTION
**Pull request by youknowriad**
This tries to resolve #10381 

In this PR, I'm adding a "banner" at the top of the post summary page when the post has no views yet.

screen shot 2017-01-18 at 15 24 17

This is still a work in progress, but I'm curious to have your thoughts on this @timmyc if you have some time. 

I preferred to add a banner instead of hiding all the stats components to avoid hiding the stats components while still fetching the stats.

### CODE REVIEW

File: client/my-sites/stats/stats-post-detail/no-views-placeholder.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Internal dependencies
+ */
+import Banner from 'components/banner';
+
+const NoViewsPlaceholder = ( { translate } ) => {
+	return (
+		<Banner
+			callToAction={ translate( 'Get more traffic!' ) }
```
Comment (a8ci18n) [inline client/my-sites/stats/stats-post-detail/no-views-placeholder.jsx]: Alternate string suggestion: 
* `translate( '<a href="%1$s">Get more traffic</a>.' )`&mdash; translations: **3**. 

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/stats/no-stats-placeholder)

**Comment by timmyc** [comment]
I definitely think we should pull in @karmatosed on the issue of the sad panda post stats page.

The main issue I see is dealing with the large components that show no data:

wordpress_com

I will look into an API fix for that issue too, but wondering if we should display messaging in those components like we do when there is no data elsewhere in stats?

stats_ _trout_bummin_ _wordpress_com

**Comment by karmatosed** [comment]
Thanks for the ping. I have a few questions to ease me into thinking about this.
Let me ask a big one first. Why do we need to show blank sections if no data? Could we somehow filter the interface to only show if there is information? 

I can imagine the answer is people want to know if nothing to show. But if everything has nothing, perhaps we could add a message or if just some have nothing, then don't show those and have a message. That's a very uninformed thought.

**Comment by timmyc** [comment]
> Why do we need to show blank sections if no data?

Excellent point!  On this particular page for Post Stats, we can have instances where the default chart up top ( last 10 days ) has no data shown, but the balance of the page with All-Time numbers do show data.

In the sad stat page scenario though, everything is empty, so maybe we should consider how other parts of calypso handle the empty data set:

banners_and_alerts_and_testimonials_ _fly_fishers_place_ _wordpress_com

We would have to wait for the all-time API call to load to determine if the post truly doesn't have any views yet, so there could be some "jumpyness" of the loading state on these components to then display an empty message.

One way @jancavan handled this before was to do an overlay over the stats page of sorts - I'd have to track down that old PR though.

**Comment by youknowriad** [comment]
@karmatosed We could filter the interface depending on whether there are views but as @timmyc noted we tend to prefere "reactive UI" https://github.com/Automattic/wp-calypso/blob/master/docs/reactivity.md and in if we were to do so, we'll have to wait for the stats request before showing the page. Maybe we could consider this as non-blocking for this particular case.

**Comment by youknowriad** [comment]
I updated the PR and I tried to limit the "loading" state as much as I could. If the stats are being requested but we already have a cached value (different from 0), I'm showing the stats modules directly in this case.

Here's how it looks like now:

screen shot 2017-01-19 at 12 28 52

File: client/my-sites/stats/stats-post-detail/index.jsx [hunk 2]
```diff
@@ -42,56 +46,79 @@ const StatsPostDetail = React.createClass( {
...
-					postViewsList={ this.props.postViewsList } />
-
-				<PostWeeks postViewsList={ this.props.postViewsList } />
+				<StatsPlaceholder isLoading={ isLoading } />
+
+				{ ! isLoading && countViews === 0 &&
+					<EmptyContent
+						title={ translate( 'Your post has not received any views yet!' ) }
+						line={ translate( 'Learn some tips to attract more visitors' ) }
+						action={ translate( 'Get more traffic!' ) }
```
Comment (a8ci18n) [inline client/my-sites/stats/stats-post-detail/index.jsx]: Alternate string suggestion: 
* `translate( '<a href="%1$s">Get more traffic</a>.' )`&mdash; translations: **3**. 

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/stats/no-stats-placeholder)

**Comment by karmatosed** [comment]
@timmyc: I really like the idea of looking to other areas. I think this works as a solution if we have to have to have it showing.

I half wonder about showing multiples, like would this be too much in a page? Or are we only showing in the first 2 sections?

**Comment by timmyc** [comment]
This is working well for me, and I think this is a good improvement over the empty stats page/tables.  @karmatosed are you okay with this going in?

**Comment by karmatosed** [comment]
👍 from me. It works and uses what we currently have, along with being better than we have.

**Comment by timmyc** [comment]
Excellent, lets rebase and get this in!

**Pull request merged**